### ISSUE TITLE
Product Request: Surface additional unverified transaction states in zcash_client_sqlite

### ISSUE THREAD

**Issue by braddmiller** [zcash/librustzcash#249]
In the ECC Wallets and Mobile SDKs, we are only ever given verified and unverified transaction states. This makes it difficult to distinguish between incoming funds that are change versus received funds and surface it through to the user. We would like to be able to show "incoming" funds in the wallet and have that be distinguished from "change" in the wallet.

Requirements:
- [ ] Incoming value for change when we request the wallet balance
- [ ] incoming value for inbound transactions when we request the wallet balance

**Comment by gmale** [comment]
One key thing to note: we can get this information in other ways but, in practice, we often want it whenever we request balance. 

For instance, on these screens, is the amount expected a result of someone sending you money? Or the result of change from a transaction that you initiated?

![Screenshot_1594322601](https://user-images.githubusercontent.com/1699841/87082001-4c0a1200-c1f8-11ea-8cbb-3227b1111f73.png)
![Screenshot_1594322595](https://user-images.githubusercontent.com/1699841/87082024-57f5d400-c1f8-11ea-8344-1a734488eabe.png)

**Comment by gmale** [comment]
Update:
Addressing this would involve creating a function similar to `get_balance_at` that allows us to query for change. Then we can subtract that from the total pending balance to infer how much is "incoming."

Such a function might look like this:

```rust
pub fn get_change_balance_at<P>(
    wdb: &WalletDb<P>,
    account: AccountId,
    anchor_height: BlockHeight,
) -> Result<Amount, SqliteClientError> {
    let balance = wdb.conn.query_row(
        "SELECT SUM(value) FROM received_notes
        INNER JOIN transactions ON transactions.id_tx = received_notes.tx
        WHERE account = ? AND spent IS NULL AND is_change = 1 AND transactions.block <= ?",
        &[account.0, u32::from(anchor_height)],
        |row| row.get(0).or(Ok(0)),
    )?;

    match Amount::from_i64(balance) {
        Ok(amount) if !amount.is_negative() => Ok(amount),
        _ => Err(SqliteClientError::CorruptedData(
            "Sum of change values in received_notes is out of range".to_string(),
        )),
    }
}
```
Note that we wold also need to update `lib.rs` on the Android and iOS sides to expose this functionality. That function would need to work with the same [height used in the getBalance function](https://github.com/zcash/zcash-android-wallet-sdk/blob/368099e1544c3d2e57f20770e62b59a41ddbc3b5/src/main/rust/lib.rs#L490) because we want to query the "total" balance but then exclude any transactions that are NOT change.

**Comment by gmale** [comment]
This is very closely related to https://github.com/zcash/zcash-android-wallet/issues/76

Fixing one will fix the other.

**Comment by pacu** [comment]
related: 
https://github.com/zcash/zcash-ios-wallet/issues/86
https://github.com/zcash/ZcashLightClientKit/issues/298

**Comment by nuttycom** [comment]
@pacu after #807, what should the status of this issue be? Does it make sense to close this now (perhaps in favor of more specific actionable items?)

**Comment by nuttycom** [comment]
This is being addressed via #900. Closing as subsumed by #900
